conan/tools/cmake/presets.py

```diff
@@ -170,8 +170,8 @@ def write_cmake_presets(conanfile, toolchain_file, generator, cache_variables,
             cmake_make_program = cmake_make_program.replace("\\", "/")
             cache_variables["CMAKE_MAKE_PROGRAM"] = cmake_make_program
 
-    if "CMAKE_POLICY_DEFAULT_CMP0091" not in cache_variables:
-        cache_variables["CMAKE_POLICY_DEFAULT_CMP0091"] = "NEW"
+        if "CMAKE_POLICY_DEFAULT_CMP0091" not in cache_variables:
+            cache_variables["CMAKE_POLICY_DEFAULT_CMP0091"] = "NEW"
```

Comment:
I think there might be some unusual but possible scenarios of cross-building from Linux to Windows that still need this definition. So it cannot be conditioned to ``platform.system() == Windows`` but it should be conditioned to ``settings.os == "Windows"``.

Reply:
Also, there are scenarios using MinGW makefiles that still use ``msvc`` as compiler and will need the Visual runtime, so this shouldn't be conditioned to the generator.

Reply:
It would be a good idea to use the same condition than the one which makes `VSRuntimeBlock` returning something not None.

`VSRuntimeBlock` is a little bit convoluted, but basically its consolidated condition seems to be:

`settings.get_safe("os") == "Windows" and settings.get_safe("compiler") in ("msvc", "clang", "intel-cc") and settings.get_safe("compiler.runtime") and settings.get_safe("build_type")`

So it could be factorized and used in `VSRuntimeBlock` and `_CMakePresets` classes.